Peer.Tests/ConnectorAPI/PeerPullRequestAPITests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using Peer.Connectors;
+using Peer.Domain;
+using Peer.Domain.Models;
+using Xunit;
+
+namespace Peer.ConnectorApi.Tests
+{
+    public abstract class PeerPullRequestAPITest<T> where T : class
+    {
+        public static void GetPullRequestsTest()
+        {
+            var mockGithubPullRequestSource = new Mock<IPullRequestSource<T>>();
+            var mocklist = new Mock<IEnumerable<PeerPullRequest>>();
+            var mockpr = new PeerPullRequest("title", "CreatedBy");
+            var listpr = new List<PeerPullRequest> { mockpr };
+            mockGithubPullRequestSource.Setup(x => x.FetchPullRequests()).Returns(Task.FromResult<IEnumerable<PeerPullRequest>>(listpr));
```

Comment:
Use ReturnsAsync instead of returning a task from result.

Reply:
Alternatively to doing this in a static method you could make this a normal method and then instead make 3 classes that inherit from this. 

So like:
public class PeerAzurePRTests : PeerPullRequestApiTests<AzDOPullRequestSource> or similar. Doesn't really matter but I think it extends more nicely when you have more tests